engine/verification/verifier/verifiers.go

```diff
@@ -0,0 +1,204 @@
+		err := verifyHeight(height, storages.Headers, chunkDataPacks, storages.Results, state, verifier)
+		if err != nil {
+			return fmt.Errorf("could not verify height %d: %w", height, err)
+		}
+	}
+
+	return nil
+}
+
+func initStorages(chainID flow.ChainID, dataDir string, chunkDataPackDir string) (
+	*badger.DB,
+	*storage.All,
+	storage.ChunkDataPacks,
+	protocol.State,
+	module.ChunkVerifier,
+	error,
+) {
+	db := common.InitStorage(dataDir)
+
+	storages := common.InitStorages(db)
+	state, err := common.InitProtocolState(db, storages)
+	if err != nil {
+		return nil, nil, nil, nil, nil, fmt.Errorf("could not init protocol state: %w", err)
+	}
+
+	chunkDataPackDB, err := storagepebble.OpenDefaultPebbleDB(chunkDataPackDir)
+	if err != nil {
+		return nil, nil, nil, nil, nil, fmt.Errorf("could not open chunk data pack DB: %w", err)
+	}
+	chunkDataPacks := storagepebble.NewChunkDataPacks(metrics.NewNoopCollector(),
+		chunkDataPackDB, storages.Collections, 1000)
+
+	verifier := makeVerifier(log.Logger, chainID, storages.Headers)
+	return db, storages, chunkDataPacks, state, verifier, nil
+}
+
+func verifyHeight(
+	height uint64,
+	headers storage.Headers,
+	chunkDataPacks storage.ChunkDataPacks,
+	results storage.ExecutionResults,
+	state protocol.State,
+	verifier module.ChunkVerifier,
+) error {
+	header, err := headers.ByHeight(height)
+	if err != nil {
+		return fmt.Errorf("could not get block header by height %d: %w", height, err)
+	}
+
+	blockID := header.ID()
+
+	if err != nil {
+		return fmt.Errorf("could not get block ID by height %d: %w", height, err)
+	}
+
+	result, err := results.ByBlockID(blockID)
+	if err != nil {
+		return fmt.Errorf("could not get execution result by block ID %s: %w", blockID, err)
+	}
+	snapshot := state.AtBlockID(blockID)
+
+	for i, chunk := range result.Chunks {
+		chunkDataPack, err := chunkDataPacks.ByChunkID(chunk.ID())
+		if err != nil {
+			return fmt.Errorf("could not get chunk data pack by chunk ID %s: %w", chunk.ID(), err)
+		}
+
+		vcd, err := convert.FromChunkDataPack(chunk, chunkDataPack, header, snapshot, result)
+		if err != nil {
+			return err
+		}
+
+		_, err = verifier.Verify(vcd)
+		if err != nil {
+			return fmt.Errorf("could not verify %d-th chunk: %w", i, err)
+		}
+	}
+	return nil
+}
+
+func makeVerifier(
+	logger zerolog.Logger,
+	chainID flow.ChainID,
+	headers storage.Headers,
+) module.ChunkVerifier {
+
+	vm := fvm.NewVirtualMachine()
+	fvmOptions := initFvmOptions(chainID, headers)
+	fvmOptions = append(
+		[]fvm.Option{fvm.WithLogger(logger)},
+		fvmOptions...,
+	)
+
+	// TODO(JanezP): cleanup creation of fvm context github.com/onflow/flow-go/issues/5249
+	fvmOptions = append(fvmOptions, computation.DefaultFVMOptions(chainID, false, false)...)
+	vmCtx := fvm.NewContext(fvmOptions...)
+
+	chunkVerifier := chunks.NewChunkVerifier(vm, vmCtx, logger)
+	return chunkVerifier
+}
+
+func initFvmOptions(chainID flow.ChainID, headers storage.Headers) []fvm.Option {
```

Comment:
can this reuse code from the standard initialization? I worry that the config will drift and we will need to maintain multiple this init in multiple files